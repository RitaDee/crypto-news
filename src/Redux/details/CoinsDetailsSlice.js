import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

const initialState = {
  list: [],
  isFectching: false,
};

export const fetchCoinsDetail = createAsyncThunk('coinsDetails/fetch', async (id) => {
  const response = await fetch(`https://api.coinstats.app/public/v1/coins${id}`);
  const data = response.json();
  console.log('Hello', data);
  return data;
});

const coinsDetailsSlice = createSlice({
  name: 'coinsDetails',
  initialState,
  extraReducers: (builder) => {
    builder
      .addCase(fetchCoinsDetail.fulfilled, (state, action) => ({
        ...state,
        isFectching: false,
        list: action.payload,
      }))
      .addCase(fetchCoinsDetail.pending, (state) => ({
        ...state,
        isFectching: true,
      }));
  },
});

export default coinsDetailsSlice.reducer;
