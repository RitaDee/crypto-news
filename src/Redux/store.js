import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './coins/CoinsSlice';
import coinsDetailsReducer from './details/CoinsDetails';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
    coinsDetails: coinsDetailsReducer,
  },
});

export default store;
