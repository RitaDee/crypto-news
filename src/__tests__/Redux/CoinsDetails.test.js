import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import { ErrorBoundary } from 'react-error-boundary';
import store from '../../Redux/store';
import CoinsDetailsSlice from '../../Redux/details/CoinsDetailsSlice';

function MyFallbackComponent({ error, resetErrorBoundary }) {
  return (
    <div>
      <h2>Something went wrong:</h2>
      <p>{error.message}</p>
      <button type="button" onClick={resetErrorBoundary}>Try again</button>
    </div>
  );
}

describe('CoinsDetailsSlice', () => {
  it('renders correctly', () => {
    const tree = renderer(
      <ErrorBoundary FallbackComponent={MyFallbackComponent}>
        <Provider store={store}>
          <CoinsDetailsSlice />
        </Provider>
      </ErrorBoundary>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

MyFallbackComponent.propTypes = {
  resetErrorBoundary: PropTypes.func.isRequired,
  error: PropTypes.shape({
    message: PropTypes.string,
  }).isRequired,
};
