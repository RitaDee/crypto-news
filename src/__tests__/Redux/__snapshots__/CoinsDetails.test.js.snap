// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinsDetailsSlice renders correctly 1`] = `
<div>
  <h2>
    Something went wrong:
  </h2>
  <p>
    Objects are not valid as a React child (found: object with keys {}). If you meant to render a collection of children, use an array instead.
  </p>
  <button
    onClick={[Function]}
    type="button"
  >
    Try again
  </button>
</div>
`;
